velox/connectors/hive/PartitionIdGenerator.cpp

```diff
@@ -37,11 +37,16 @@ PartitionIdGenerator::PartitionIdGenerator(
         exec::VectorHasher::create(inputType->childAt(channel), channel));
   }
 
-  std::vector<std::string> partitionKeyNames;
   std::vector<TypePtr> partitionKeyTypes;
+  std::vector<std::string> partitionKeyNames;
   for (auto channel : partitionChannels_) {
-    partitionKeyNames.push_back(inputType->nameOf(channel));
+    VELOX_CHECK(
+        exec::VectorHasher::typeKindSupportsValueIds(
+            inputType->childAt(channel)->kind()),
+        "Unsupported partition type: {}.",
+        inputType->childAt(channel)->name());
```

Comment:
name() -> toString() to provide more details for complex types (ARRAY(BIGINT) vs. ARRAY)

Reply:
Good to know!